src/main/scala/scalaz/reactive/Event.scala

```diff
@@ -1,11 +1,37 @@
 package scalaz.reactive
+import scalaz.Monoid
+import scalaz.zio.{ IO, RTS }
 
-case class Event[+A](value: Future[Reactive[A]]) extends AnyVal {
+case class Event[A](value: Future[Reactive[A]]) extends RTS { self =>
+
+  def merge(
+    v: Event[A]
+  ): Event[A] = { // if Event[+A]: covariant type A occurs in contravariant position
+
+    val winner: IO[Nothing, Either[(Time, Reactive[A]), (Time, Reactive[A])]] =
+      self.value.force.map(Left(_)).race(v.value.force.map(Right(_)))
+
+    val io: IO[Nothing, Future[Reactive[A]]] = winner.map(_ match {
+      case Left((t, r)) =>
+        Future((t, Reactive(r.head, r.tail)))
+      case Right((t, r)) =>
+        Future((t, Reactive(r.head, r.tail)))
+    })
+
+    Event(unsafeRun(io)) // That is not good
+  }
 
   def map[B](f: A => B): Event[B] =
     Event(value.map(_.map(f)))
 
-  // TODO monad
+  implicit def monoidEvent[E, A]: Monoid[Event[A]] =
+    new Monoid[Event[A]] {
+
+      override def zero: Event[A] = Event(Future.Never)
```

Comment:
If we make Future = IO, then this is a good one because we would need to make an IO which never resolves and always loses race operations, maybe, for now, an IO with a fiber that is always blocked? we can analyze efficiency later

Reply:
@FrancoAra IO provides IO.never - _Returns a action that will never produce anything._

Reply:
@hienqnguyen I guess we should do the racing in the Future's `+` and end up with what you say.